docs/framework/pseudoconstant.md

```diff
@@ -0,0 +1,86 @@
+    -   Each of these tables has its own api for manipulating the values.
+-   A few option lists are stored as *enum* values, for example the contact `preferred_mail_format` field is of type *enum.*
+-   A few others are not in the database at all, but hard-coded as php arrays. The list `{0: 'No', 1: 'Yes'}` is an example of this for  boolean fields.
+
+### Option List Retrieval
+
+As of CiviCRM 4.4 (and to a limited extent 4.3) there are option retrieval methods which bridge these different ways of storage and return options in a predictable way for every field in the database.
+
+!!! note **Historical note**
+Prior to 4.4, the `CRM_*_Pseudoconstant` classes contained 1 function per option list. Those single-purpose methods are deprecated and in 4.4 many have been removed. The 'constant' api is therefore also *deprecated* in CiviCRM 4.3 and 4.4 in favor of the api.getoptions method.
+
+| Method | Version Added | Purpose | Example |
+| --- | --- | --- | --- |
+| `CRM_Core_Pseudoconstant::getName` | 4.4 | Retrieve an option name given its key | `$baoName = "CRM_Core_BAO_Address"; $key = 1228; $name = CRM_Core_Pseudoconstant::getName($baoName, 'country_id', $key); echo $name; // 'US'` |
+| `CRM_Core_Pseudoconstant::getLabel` | 4.4 | Retrieve an option label given its key  | `$baoName = "CRM_Core_BAO_Address"; $key = 1228; $label = CRM_Core_Pseudoconstant::getLabel($baoName, 'country_id', $key); echo $label; // 'United States'` |
+| `CRM_Core_Pseudoconstant::getKey` | 4.4 | Retrieve an option key given its name  | `$baoName = "CRM_Core_BAO_Address"; $iso = 'US' ; // note that iso-codes are used as machine names for countries $key = CRM_Core_Pseudoconstant::getKey($baoName, 'country_id', $iso); echo $key; // 1228` |
+| `CRM_*_BAO_*::buildOptions` | 4.4 | Retrieve a list of options for a field | `CRM_Contact_BAO_Contact::buildOptions('gender_id');` Note: accepts additional params related to context. See below. |
+| `api.getoptions` | 4.3 | In 4.3 - retrieve option list for a field - works for many but not all fields. In 4.4 - Api wrapper around `CRM_*_BAO_*::buildOptions ` - works for all fields and passes additional params to buildOptions. | `civicrm_api3('contact', 'getoptions', array('field' => 'gender_id'));` Note: accepts additional params in 4.4. See below. |
+
+#### Context
+
+The format of an option list needs to be slightly different depending on the context you're working in. For example if you have machine names as input and need to store the values in the database, you need the list to contain ids and names instead of ids and translated labels. The BAO `buildOptions` method (and the `api.getoptions` wrapper) accept a 'context' string which controls formatting. Quoting from the `CRM_Core_BAO::buildOptionsContext` code:
+
+ - `'get` -  "All options are returned, even if they are disabled.  Labels are translated.",
+ - `create` - "Options are filtered appropriately for the object being created/updated. Labels are translated.",
```

Comment:
@mattwire good points i think looking at this https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/PseudoConstant.php#L213 suggests to me it would return only enabled options